### ISSUE TITLE
Add backslashed single quotes to grammar

### ISSUE THREAD

**Issue by kislikjeka98** [DerekStride/tree-sitter-sql#262]
It breaks if use single quoted string with backslash escaped single quote, for example 'Author\'s name'.
But mysql and postgres accept it
https://dev.mysql.com/doc/refman/8.4/en/string-literals.html
https://www.postgresql.org/docs/current/sql-syntax-lexical.html

I think it can be fixed by changing single_quote grammar to this 
```
{
	"_single_quote_string": {
		"type": "SEQ",
		"members": [
			{
				"type": "PATTERN",
				"value": "([uU]&)?'(''|[^'\\]|\\.)*'"
			},
			{
				"type": "REPEAT",
				"content": {
					"type": "PATTERN",
					"value": "'(''|[^'\\]|\\.)*'"
				}
			}
		]
	}
}

```

**Comment by matthias-Q** [comment]
This change is not enough, unfortunately. I have tested it and it breaks some tests (`COPY FROM STDIN`, because there a backslash can be set as an escape char)

**Comment by kislikjeka98** [comment]
In my mind COMENT used more frequiently than copy statements with EXCAPE '\'

**Comment by matthias-Q** [comment]
Correct, but the parser work in all cases. I would really not like to remove functionality.
